tests/common/threads/test_history.py

```diff
@@ -0,0 +1,130 @@
+import pytest
+import json
+
+from tests.test_utils import MockCache, MockContext, MockMessage
+from tests.test_utils import mock_gql_query
+from tests.test_utils import (
+    assert_message_content,
+    assert_message_reaction,
+)
+from bot.config import (
+    YES_EMOJI,
+    NO_EMOJI,
+)
+
+from bot.common.threads.history import (
+    DisplayHistoryStep,
+    GetContributionsCsvPromptStep,
+    GetContributionsCsvPromptStepEmoji,
+)
+
+from bot.common.threads.thread_builder import StepKeys
+
+default_contributions = [
+    {
+        # TODO: extract these
+        "date_of_submission": "2022-09-21",
+        "date_of_engagement": "2022-09-21",
+        "name": "unit tests 1",
+        "status": {"name": "great"},
+    },
+    {
+        "date_of_submission": "2022-09-22",
+        "date_of_engagement": "2022-09-22",
+        "name": "unit tests 2",
+        "status": {"name": "ok"},
+    },
+]
+
+
+@pytest.mark.asyncio
+async def test_display_history_step_user_not_found(mocker):
+    cache = MockCache()
+    context = MockContext()
+    message = MockMessage()
+    user_id = "1"
+    history_step = DisplayHistoryStep(None, cache, None, context)
+
+    mock_gql_query(mocker, method="fetch_user_by_discord_id", returns=None)
+    (sent_message, tmp) = await history_step.send(message, user_id)
+
+    # Check prompt for joining
+    expected_message = DisplayHistoryStep.onboard_prompt_content
+    assert_message_content(sent_message, expected_message)
+
+
+@pytest.mark.asyncio
+async def test_display_history_step_no_contributions(mocker):
+    cache = MockCache()
+    context = MockContext()
+    message = MockMessage()
+    user_id = "1"
+    history_step = DisplayHistoryStep(None, cache, None, context)
+
+    mock_gql_query(mocker, "fetch_user_by_discord_id", returns={"id": "1"})
+    mock_gql_query(mocker, "get_guild_by_discord_id", returns={"id": "1"})
+    mock_gql_query(mocker, "get_contributions_for_guild", returns=None)
+    (sent_message, tmp) = await history_step.send(message, user_id)
+    assert (
+        DisplayHistoryStep.no_contributions_content == sent_message.content
+    ), "expected no contributions message"
+
+
+@pytest.mark.asyncio
+async def test_display_history_step(mocker):
+    cache = MockCache()
+    context = MockContext()
+    message = MockMessage()
```

Comment:
To be dryer you could probably wrap these objects in a fixture to make them reusable across tests.